sources/packages/backend/apps/queue-consumers/src/services/cas-invoice-batch/cas-invoice-batch.service.ts

```diff
@@ -0,0 +1,250 @@
+        invoiceDetail.casDistributionAccount = account;
+        invoiceDetail.valueAmount = disbursedAward.effectiveAmount;
+        invoiceDetail.creator = this.systemUsersService.systemUser;
+        return invoiceDetail;
+      });
+      invoiceDetails.push(...awardInvoiceDetails);
+    }
+    return invoiceDetails;
+  }
+
+  /**
+   * Retrieves all active distribution accounts to generate invoice details.
+   * @param entityManager entity manager to be used for the query.
+   * @returns list of active distribution accounts.
+   */
+  private async getActiveDistributionAccounts(
+    entityManager: EntityManager,
+  ): Promise<CASDistributionAccount[]> {
+    return entityManager.getRepository(CASDistributionAccount).find({
+      select: {
+        id: true,
+        awardValueCode: true,
+        offeringIntensity: true,
+        operationCode: true,
+      },
+      where: { isActive: true },
+    });
+  }
+
+  /**
+   * Retrieves the disbursement receipts that have no valid invoice associated,
+   * filtering only the receipts with BC Grant values and valid CAS supplier.
+   * @param entityManager entity manager to be used for the query.
+   * @returns list of disbursement receipts without valid invoice.
+   */
+  private async getPendingReceiptsForInvoiceBatch(
+    entityManager: EntityManager,
+  ): Promise<DisbursementReceipt[]> {
+    const existsInvoice = entityManager
+      .getRepository(CASInvoice)
+      .createQueryBuilder("casInvoice")
+      .select("1")
+      .where("casInvoice.disbursementReceipt.id = disbursementReceipt.id")
+      .getSql();
+    return entityManager
+      .getRepository(DisbursementReceipt)
+      .createQueryBuilder("disbursementReceipt")
+      .select([
+        "disbursementReceipt.id",
+        "disbursementSchedule.id",
+        "studentAssessment.id",
+        "offering.id",
+        "offering.offeringIntensity",
+        "application.id",
+        "student.id",
+        "casSupplier.id",
+        "casSupplier.supplierNumber",
+        "disbursementValue.id",
+        "disbursementValue.valueCode",
+        "disbursementValue.effectiveAmount",
+      ])
+      .innerJoin(
+        "disbursementReceipt.disbursementReceiptValues",
+        "disbursementReceiptValue",
+      )
+      .innerJoin(
+        "disbursementReceipt.disbursementSchedule",
+        "disbursementSchedule",
+      )
+      .innerJoin("disbursementSchedule.disbursementValues", "disbursementValue")
+      .innerJoin("disbursementSchedule.studentAssessment", "studentAssessment")
+      .innerJoin("studentAssessment.offering", "offering")
+      .innerJoin("studentAssessment.application", "application")
+      .innerJoin("application.student", "student")
+      .innerJoin("student.casSupplier", "casSupplier")
+      .where("disbursementReceiptValue.grantType = :grantType", {
+        grantType: BC_TOTAL_GRANT_AWARD_CODE,
+      })
+      .andWhere("disbursementReceiptValue.grantAmount > 0")
```

Comment:
What about considering using the these values from disbursement_receipts table instead of going into disbursement_receipt_values ? 
![image](https://github.com/user-attachments/assets/322716e0-fc8f-4058-b81a-856664869575)

Do we think using the `disbursementValue.valueType = BCSG` will be more accurate?

Reply:
If BCSG was not available I would be more open to using the `total_disbursed_grant_amount`, since we have BCSG it does seem more accurate, IMO.
BCSG represents the exact SUM of provincial grants generated by SIMS and the receipt sent back.

Reply:
I see that we can avoid joining the disbursement_receipt_values if we can use `total_disbursed_grant_amount` here. But not a blocker. Let me know if I am missing something.

Reply:
I do understand it, but to me was clear that the check must happen using the BCSG and that would be my recommendation also. I will ask the business if they can ensure the goal will be achieved in the same way I will change.